lib/src/components/base/service.dart

```diff
@@ -0,0 +1,70 @@
+import 'package:grpc/grpc.dart';
+
+import '../../gen/common/v1/common.pb.dart';
+import '../../gen/component/base/v1/base.pbgrpc.dart';
+import '../../resource/manager.dart';
+import '../../utils.dart';
+import 'base.dart';
+
+class BaseService extends BaseServiceBase {
+  final ResourceManager _manager;
+
+  BaseService(this._manager);
+
+  Base _fromManager(String name) {
+    try {
+      return _manager.getResource(Base.getResourceName(name));
+    } catch (e) {
+      throw GrpcError.notFound(e.toString());
+    }
+  }
```

Comment:
I like how you named this, omitting `base`. I will copy in other services.

I've been thinking how we can abstract some of the boiler plate code thats essentially copied between each service, maybe a service `mixin` would be a good move to abstract this out and include there.

Reply:
Ya same. Tried abstracting it and ran into issues around not having abstract static functions. Renamed to be just `_fromManager` to make it easier to copy and paste until we figure out a way to refactor it lol

Reply:
SGTM, I'm down for finding the right abstractions after we have things fleshed out.